### PR TITLE
(TK-248) Wrapping intermediate comidi routes with middleware

### PR DESCRIPTION
**Pull request by scottyw**
Add a new wrap-routes function that is analagous to compojure's that wraps each leaf handler with a middleware.

### CODE REVIEW

**Comment by cprice404** [comment]
yay

**Comment by cprice404** [comment]
@rlinehan can I solicit you as a second review on this one as well :)

**Comment by rlinehan** [comment]
suuuuuure. it's not gonna be until next week though.

File: test/puppetlabs/comidi_test.clj [hunk 1]
```diff
@@ -288,3 +288,43 @@
...
+        left-routes (context "/left" aa-route bb-route)
+        middle-routes (context "/middle" cc-route dd-route)
+        right-routes (context "/right" ee-route ff-route)
+        handler (-> (routes left-routes middle-routes right-routes) routes->handler)]
+    (is (= (:body (handler {:uri "/left/aa" :request-method :get})) "aa!"))
+    (is (= (:body (handler {:uri "/left/bb" :request-method :post})) "bb!"))
+    (is (= (:body (handler {:uri "/middle/cc" :request-method :get})) "cc!"))
+    (is (= (:body (handler {:uri "/middle/dd" :request-method :delete})) "dd!"))
+    (is (= (:body (handler {:uri "/right/ee" :request-method :get})) "ee!"))
+    (is (= (:body (handler {:uri "/right/ff" :request-method :delete})) "ff!"))
```
Comment (rlinehan) [inline test/puppetlabs/comidi_test.clj]: I think this test would be a bit clearer if there were two `testing` blocks, one above this block of assertions saying that it's testing the routes before any middleware is applied, and another around the next set of assertions, saying that it's the routes after `wrap-routes` is used.

**Comment by rlinehan** [comment]
:+1: left one small comment about adding some `testing` lines, but aside from that this looks good.

File: test/puppetlabs/comidi_test.clj (hunk 1 above)
Comment (scottyw) [inline test/puppetlabs/comidi_test.clj]: Good call, I just made the change. Thanks!

**Comment by rlinehan** [comment]
:shipit:

**Pull request merged**